### PR TITLE
exclude fusions which have NAfrom AGFusion clinical run

### LINKED ISSUE #158: agfusion clinical failing due to NA
### Description of the bug

I got the following error for AGFUSION_CLINICAL:

```
/usr/local/lib/python3.10/dist-packages/pandas/core/computation/expressions.py:21: UserWarning: Pandas requires version '2.8.4' or newer of 'numexpr' (version '2.8.1' currently installed).
  from pandas.core.computation.check import NUMEXPR_INSTALLED
/usr/local/lib/python3.10/dist-packages/pandas/core/arrays/masked.py:60: UserWarning: Pandas requires version '1.3.6' or newer of 'bottleneck' (version '1.3.2' currently installed).
  from pandas.core import (
WARNING:matplotlib:Matplotlib created a temporary cache directory at /scratch/tempobot/matplotlib-xrf651__ because the default path (/home/tempobot/.config/matplotlib) is not a writable directory; it is highly recommended to set the MPLCONFIGDIR environment variable to a writable directory, in particular to speed up the import of Matplotlib and to better support multiprocessing.
INFO:matplotlib.font_manager:generated new fontManager
/usr/local/lib/python3.10/dist-packages/matplotlib/projections/__init__.py:63: UserWarning: Unable to import Axes3D. This may be due to multiple versions of Matplotlib being installed (e.g. as a system package and as a pip package). As a result, the 3D projection is not available.
  warnings.warn("Unable to import Axes3D. This may be due to multiple versions of "
2025-05-20 03:18:24,948 - AGFusion - WARNING - Output directory mysample already exists! Overwriting...
WARNING:AGFusion:Output directory mysample already exists! Overwriting...
2025-05-20 03:18:24,948 - AGFusion - INFO - Read 1 fusions from the file.
INFO:AGFusion:Read 1 fusions from the file.
2025-05-20 03:18:25,300 - AGFusion - WARNING - You provided a transcript ID as well as specified --noncanonical flag. Will process all the gene's transcripts.
WARNING:AGFusion:You provided a transcript ID as well as specified --noncanonical flag. Will process all the gene's transcripts.
2025-05-20 03:18:25,366 - AGFusion - ERROR - No Ensembl ID found for NA! Check its spelling and if you are using the right genome build.
ERROR:AGFusion:No Ensembl ID found for NA! Check its spelling and if you are using the right genome build.
awk: cannot open mysample/*/*.fusion_transcripts.csv (No such file or directory)
```
the input file has an NA for gene3 ids
```
$ cat /juno/cmo/ccs/noronhaa/forte_dev/bam2fq_bugfix/workspace/work2/77/1a85a0a73d856c077fecc40ae8c29a/mysample_metafusion_cluster.unfiltered.clinical.cff | cut -f 14,16,19,21,38,39
gene5_renamed_symbol	gene3_renamed_symbol	reann_gene5_symbol	reann_gene3_symbol	gene5_transcript_id	gene3_transcript_id
YAP1	.	YAP1	NA	ENST00000282441	NA
```

### Command used and terminal output

```console

```

### Relevant files

_No response_

### System information

_No response_

### PR DESCRIPTION
**Pull request by pintoa1-mskcc**
## PR checklist

- [x] Removes any fusion with NA as one of the gene pairs from expanded clinical agfusion runs. When agfusion hits a fusion with NA as one of the genes, it will error out and fails. Closes #158 
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] `CHANGELOG.md` is updated.

### CODE REVIEW

File: modules/local/add_flags/main.nf [hunk 1]
```diff
@@ -33,7 +33,7 @@ process ADD_FLAG {
     cat *_metafusion_cluster.unfiltered.cff \\
         | awk 'FNR <= 1' > header.txt
     cat *_metafusion_cluster.unfiltered.cff \\
-        | grep -iFwf $clinical_genes > tmp_clinicalgenes.txt
+        | grep -iFwf $clinical_genes | awk '\$20 != "NA" && \$22 != "NA"' > tmp_clinicalgenes.txt
```
Comment (anoronh4) [inline modules/local/add_flags/main.nf]: can you also add an `|| test $? = 1` to the end of this line? when the grep operation has zero results it seems to be erroring out.

File: modules/local/agfusion/batch/main.nf [hunk 2]
```diff
@@ -14,8 +14,7 @@ process AGFUSION_BATCH {
     path(pyensembl_cache)
 
     output:
-    tuple val(meta), path("${prefix}/**")                    , emit: fusions_annotated
-    tuple val(meta), path("${prefix}.fusion_transcripts.csv"), emit: fusion_transcripts_csv
+    tuple val(meta), path("${prefix}/**")                    , emit: fusions_annotated, optional: true
```
Comment (anoronh4) [inline modules/local/agfusion/batch/main.nf:17]: if we `mkdir -p ${prefix}/` does the folder still have to be `optional: true`? i'm just not sure if the contents of this folder is one of the checks that the prod team is checking. might be good to check with @huyu335 about this.

Comment (pintoa1-mskcc) [reply modules/local/agfusion/batch/main.nf:17]: From the list yu hu gave me, I think we only need the agfusion directory, not the subdiretories as those are unique to the sample

Comment (anoronh4) [reply modules/local/agfusion/batch/main.nf:17]: so looks like you already check with yu hu, great! i'm going to approve.

**Review by anoronh4 — APPROVED** [review]

**Review by carynhale — APPROVED** [review]
